nikola/plugins/compile/markdown/__init__.py

```diff
@@ -43,40 +44,56 @@
 from nikola.utils import makedirs, req_missing, write_metadata
 
 
+class ThreadLocalMarkdown(threading.local):
+    """Convert to markdown using per-thread Markdown objects."""
+
+    def __init__(self, extensions):
+        self.markdown = Markdown(extensions=extensions, output_format="html5")
```

Comment:
Here’s some test code:

```py
import threading
from markdown import Markdown

class ThreadLocalMarkdown(threading.local):
    """Convert to markdown using per-thread Markdown objects."""

    def __init__(self, extensions):
        self.markdown = Markdown(extensions=extensions, output_format="html5")
        self.counter = 0

    def convert(self, data):
        """Convert data to HTML and reset internal state."""
        result = self.markdown.convert(data)
        self.markdown.reset()
        self.counter += 1
        print(self.counter)
        return result

    def show(self):
        print(self.counter)

converter = ThreadLocalMarkdown([])

def run_convert():
    converter.convert("Hello, **world!**")

threads = [threading.Thread(None, run_convert) for i in range(10)]
[t.start() for t in threads]
converter.show()
```

says a bunch of ones and a zero at the end, as expected.